### PR TITLE
Treca zadaca

### PR DESCRIPTION
**Pull request by MVuko123**
317843930_888504765661829_6811789427734599088_n
318033668_473493671438959_4071624984823453071_n
317931775_3095511400741133_8369860918887629737_n

### CODE REVIEW

File: app/src/main/java/agency/five/codebase/android/movieapp/ui/component/FavoriteButton.kt [hunk 1]
```diff
@@ -20,7 +20,7 @@ import androidx.compose.ui.unit.dp
 fun FavoriteButton(
     isFavorite: Boolean = false,
     modifier: Modifier = Modifier,
-    favoriteClick: (Boolean) -> Unit,
+    favoriteClick: () -> Unit,
```
Comment (anvukelic) [inline app/src/main/java/agency/five/codebase/android/movieapp/ui/component/FavoriteButton.kt:23]: you need to pass id at least when you click favorite

File: app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                _upcomingMovieHomeViewState.value = homeScreenMapper.toHomeMovieCategoryViewState(
+                    movieCategories = upcomingCategory,
+                    selectedMovieCategory = MovieCategory.UPCOMING_TODAY,
+                    movies = it
+                )
+            }
+        }
+    }
+
+
```
Comment (anvukelic) [inline app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeViewModel.kt]: redundant empty line

File: app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                _popularMoviesHomeViewState.value = homeScreenMapper.toHomeMovieCategoryViewState(
+                    movieCategories = popularCategory,
+                    selectedMovieCategory = MovieCategory.getByOrdinal(labelId)!!,
+                    movies = it
+                )
+            }
+        }
+    }
+
+    fun toggleNowPlaying(labelId : Int){
```
Comment (anvukelic) [inline app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeViewModel.kt]: you're not toggling movies, you're fetching a new list - toggle means that some value is changed from 0 to 1 and vice versa. Please fix naming

File: app/src/main/java/agency/five/codebase/android/movieapp/ui/moviedetails/MovieDetailsScreen.kt [hunk 4]
```diff
@@ -110,10 +120,12 @@ fun MovieDetailsBanner(
...
-            val onFavoriteClick = !movieDetailsViewState.isFavorite
-            FavoriteButton(isFavorite = movieDetailsViewState.isFavorite) {
-                onFavoriteClick
-            }
+            FavoriteButton(
+                isFavorite = movieDetailsViewState.isFavorite,
+                favoriteClick = {
+                    onFavoriteClick(movieDetailsViewState.id
+                    )
+                })
```
Comment (anvukelic) [inline app/src/main/java/agency/five/codebase/android/movieapp/ui/moviedetails/MovieDetailsScreen.kt]: formatting

**Review by anvukelic — CHANGES_REQUESTED** [review]

**Comment by anvukelic** [comment]
Good homework, small changes and you're ready to go

File: app/src/main/java/agency/five/codebase/android/movieapp/ui/component/FavoriteButton.kt (hunk 1 above)
Comment (MVuko123) [reply app/src/main/java/agency/five/codebase/android/movieapp/ui/component/FavoriteButton.kt:23]: I give it the movie id when I use it in MovieDetailsScreen on line 126, in HomeScreen on line 193

**Review by anvukelic — APPROVED** [review]

**Pull request merged**